### DISCUSSION TITLE
Keyboard shortcuts for saved replies do not work the first time

### DISCUSSION THREAD

**Discussion by AlCalzone** [community/community#77428] [Accessibility]
### Select Topic Area

Bug

### Body

When accessing saved replies using keyboard shortcuts (<kbd>CTRL</kbd> + <kbd>.</kbd> -> <kbd>CTRL</kbd> + <kbd>1</kbd>), actually selecting the reply using the second shortcut does not work the first time. The dialog needs to be closed and reopened for this to work.
This breaks every time after navigating to a new page, or reloading a page with a text field.

To reproduce:
1. Open a text field, like on issues/PRs/discussions
2. Press <kbd>CTRL</kbd> + <kbd>.</kbd>
3. Press <kbd>CTRL</kbd> + <kbd>1</kbd> to select the first reply
4. (... nothing happens ...)
5. Close the dialog
6. Press <kbd>CTRL</kbd> + <kbd>.</kbd>
7. Press <kbd>CTRL</kbd> + <kbd>1</kbd> to select the first reply
8. (... saved reply is inserted ...)

**Comment by jeana-redhat** [comment]
This started happening when the **Saved replies** menu was changed from a drop-down to a pop-up for me. Really bad usability regression.

**Comment by ghostinhershell** [comment]
Hey there @AlCalzone and @jeana-redhat, 

Thanks so much for sharing your feedback and reporting this issue. I'll test this out on my end and follow-up with the Accessibility team so that they can review.

**Comment by jeana-redhat** [comment]
@ghostinhershell thanks! 

In case it's helpful info, for me this is in Firefox on Linux

PS great username :grinning:

**Comment by jeana-redhat** [comment]
Today, this is fixed for me :tada: 

I still have the new pop-up UI, but now:
1. Hit <kbd>Ctrl</kbd> + <kbd>.</kbd>
2. Saved replies menu pop up.
3. Hit <kbd>Ctrl</kbd> + <kbd>1</kbd>
4. First saved reply is populated into the comment field and the pop-up disappears

This functions correctly, even on the first attempt (no need for close/reopen workaround as before).

Cheers to whoever got the fix in so quickly! Hope the original reporter is also seeing it working for them :crossed_fingers:

**Comment by AlCalzone** [comment]
Looks like this is solved. Thanks!

**Comment by AlCalzone** [comment]
@ghostinhershell recently this is broken for me again.

**Comment by jeana-redhat** [comment]
Yep, same for me :thinking:

**Comment by cehfisher** [comment]
Hi @AlCalzone - thank you for providing this feedback! We wanted to follow up on this reported bug. Is this still an issue? If it is resolved, could you please close out this issue? If it is not resolved, we can file this as a new issue for the Accessibility team to review. Thank you! (cc @queenofcorgis)

**Reply by AlCalzone** [reply]
It's still happening sometimes. I **think** this might be a race condition. I noticed that this happens when the saved replies are loaded and appear after a short delay after the loading spinner.
When they appear immediately, it usually works.

**Reply by cehfisher** [reply]
Thank you for providing this additional feedback! We will file this with our Accessibility team so that it can be reviewed and will follow-up here with any additional questions or updates. Thanks again!

**Reply by AlCalzone** [reply]
Any information I can collect for you when it happens, that can help fix this? I'm running into this regularly, but I haven't really been able to make sense of it.

**Comment by JoeEasy254** [comment]
#126107

**Comment by AlCalzone** [comment]
I hate to say it, but this feature is broken again for me. The CTRL+number shortcuts don't work at all in the current Firefox right now.

**Reply by jeana-redhat** [reply]
Noticed this in Firefox yesterday as well.

**Reply by cehfisher** [reply]
Sorry to hear about this regression! We will update the internal issue and get back with any updates or follow-up questions.

**Reply by AlCalzone** [reply]
@cehfisher I was wondering if you had some update to share. As it stands, the shortcuts are still broken for me.

**Reply by cehfisher** [reply]
Apologies, there's no update yet, @AlCalzone. However, the accessibility team is actively investigating this keyboard shortcut issue along with others. We'll check back in with updates as soon as we have more information.

**Reply by diminutivesloop** [reply]
This is also broken for me in Firefox and unlike the original report trying a second time still doesn't work. I did try assigning the ctrl+<number> shortcut to one of my browser extensions and it worked so there shouldn't be anything else causing a conflict.

**Reply by AlCalzone** [reply]
Just to be clear, they don't work at all for me either. No matter how often I open the dialog.

**Comment by tejisin** [comment]
Been broken a while and still is, sadly :(
